framework/src/modules/chain/dpos/delegates_info.js

```diff
@@ -82,21 +82,29 @@ class DelegatesInfo {
 		 * update anything in the accounts.
 		 */
 		if (block.height === 1) {
+			const round = 1;
+			await this.delegatesList.createRoundDelegateList(round, tx);
+
 			return false;
 		}
 
 		await this._updateProducedBlocks(block, undo, tx);
 
 		// Perform updates that only happens in the end of the round
 		if (this._isLastBlockOfTheRound(block)) {
+			const round = this.slots.calcRound(block.height);
+
 			if (undo) {
 				/**
 				 * If we are reverting the block, new transactions
 				 * can change vote weight of delegates, so we need to
 				 * invalidate the cache for the next rounds.
 				 */
-				const round = this.slots.calcRound(block.height);
 				await this.delegatesList.deleteDelegateListAfterRound(round, tx);
+			} else {
+				// Create round delegate list
+				const nextRound = round + 1;
+				await this.delegatesList.createRoundDelegateList(nextRound, tx);
```

Comment:
I think we have to create the round delegate list after applying rewards fees and updating the votedDelegateVoteWeights after await Promise all.

Reply:
Does the order matter? We can also create the list before this Promise.all?

Reply:
ah that's true.
delegate weight changes after this `Promise.all`, so the round delegate list should be generated after that